### ISSUE TITLE
Extlib.Structures seems to no longer work with QC

### ISSUE THREAD

**Issue by mikkelmilo** [QuickChick/QuickChick#191]
After upgrading to coq 8.11.0 I have trouble getting even the simplest examples to run, and it seems to be ext-lib causing the problems. With this program:
```
From QuickChick Require Import QuickChick.
Import MonadNotation. 

Definition a : G nat := 
  returnGen 1.
Definition b : G nat := 
  v <- a ;;
  returnGen v.

Sample a.
Sample b.
Sample (liftM Some a).
```

the first sample works fine, but the two latter gives me the error 
```
Could not compile test program
File "QuickChicke84d84.ml", line 705, characters 4-5:
Error: The type of this expression, '_weak1 -> '_weak2 -> int,
       contains type variables that cannot be generalized
Command exited with code 2.
```
if I replace the "<-" notation with bindGen, `Sample b` works fine.

This is my opam switch:
```
# Packages matching: installed
# Name                   # Installed               # Synopsis
base-bigarray            base
base-threads             base
base-unix                base
conf-findutils           1                         Virtual package relying on findutils
conf-m4                  1                         Virtual package relying on m4
coq                      8.11.0                    pinned to version 8.11.0
coq-bignums              8.11.0                    Bignums, the Coq library of arbitrary large numbers
coq-equations            1.2.1+8.11                A function definition package for Coq
coq-ext-lib              0.11.1                    A library of Coq definitions, theorems, and tactics
coq-mathcomp-ssreflect   1.10.0                    Small Scale Reflection
coq-metacoq              1.0~alpha2+8.11           A meta-programming framework for Coq
coq-metacoq-checker      1.0~alpha2+8.11           Specification of Coq's type theory and reference checker implementation
coq-metacoq-erasure      1.0~alpha2+8.11           Implementation and verification of an erasure procedure for Coq
coq-metacoq-pcuic        1.0~alpha2+8.11           A type system equivalent to Coq's and its metatheory
coq-metacoq-safechecker  1.0~alpha2+8.11           Implementation and verification of safe conversion and typechecking algorithms for Coq
coq-metacoq-template     1.0~alpha2+8.11           A quoting and unquoting library for Coq in Coq
coq-metacoq-translations 1.0~alpha2+8.11           Translations built on top of MetaCoq
coq-quickchick           1.3.0                     Randomized Property-Based Testing Plugin for Coq
coq-simple-io            1.3.0                     IO monad for Coq
coq-stdpp                dev.2020-04-06.1.087e4ccf std++ is an extended "Standard Library" for Coq
dune                     2.4.0                     Fast, portable, and opinionated build system
menhir                   20200211                  An LR(1) parser generator
menhirLib                20200211                  Runtime support library for parsers generated by Menhir
menhirSdk                20200211                  Compile-time library for auxiliary tools related to Menhir
num                      1.3                       The legacy Num library for arbitrary-precision integer and rational arithmetic
ocaml                    4.07.1                    The OCaml compiler (virtual package)
ocaml-base-compiler      4.07.1                    Official release 4.07.1
ocaml-config             1                         OCaml Switch Configuration
ocamlbuild               0.14.0                    OCamlbuild is a build system with builtin rules to easily build most OCaml projects.
ocamlfind                1.8.1                     A library manager for OCaml
```
What could be the problem here?

**Comment by lemonidas** [comment]
I just replicated this on my machine.

**TLDR / Quick steps fix for your setup:**
- Clone QuickChick (branch 8.11)
- Comment lines 158 in src/QuickChick.mlg back in (the line that executes "touch <file>.mli"
- make install

This would be a quick fix for your problem before a new version is released.

**Context:**
There used to be an issue with Coq's extraction producing too strong .mli files where the .ml doesn't satisfy the interface. Therefore QuickChick deletes that .mli. But sometimes without an interface to constrain functions, OCaml will sometimes complain about weak variables that cannot be generalized. That's why QuickChick used to create an empty .mli file.

@liyishuai 
I see you were the one who commented this out as part of a fix for the decimal/int problem (15f200c0dc4b97dcb073ab110b34728e8f5a3a16) . Was that intentional or should we put the "touch" back in?

**Comment by liyishuai** [comment]
I forgot the specific reason for commenting this out. Adding it back on my fork and running CI now.

**Comment by liyishuai** [comment]
Fixed now in `coq-quickchick.8.11.dev`, and soon in version 1.3.1.

**Comment by mikkelmilo** [comment]
Thanks for the quick replies, and the fix!
